app/controllers/pusher_controller.rb

```diff
@@ -4,12 +4,26 @@ class PusherController < ApplicationController
   # stop rails CSRF protection for this action
   protect_from_forgery except: :auth
 
+  # POST /pusher/auth
+  # @argument channel_name [String] The name of the pusher channel.
+  # @argument socket_id [String] The id of the pusher socket.
+  #
+  # Returns an auth token from Pusher that looks like this
+  # {"auth": "some-auth-key"}
+  #
+  # If there is an error, the response will look like this
+  # {"channel_name":["can't be blank"],"socket_id":["can't be blank"]}
   def auth
-    if current_user
```

Comment:
Because PusherController is a child of ApplicationController, it is protected by the `before_action :if_not_signed_in`, so there will always be a current user and we don't need to check it in this endpoint.